run_linux.go

```diff
@@ -409,7 +413,7 @@ func runSetupBuiltinVolumes(mountLabel, mountPoint, containerDir string, builtin
 	return mounts, nil
 }
 
-func (b *Builder) setupMounts(mountPoint string, spec *specs.Spec, bundlePath string, optionMounts []specs.Mount, bindFiles map[string]string, builtinVolumes, volumeMounts []string, shmSize string, namespaceOptions define.NamespaceOptions, secrets map[string]string, runFileMounts []string) (runMountTargets []string, err error) {
+func (b *Builder) setupMounts(mountPoint string, spec *specs.Spec, bundlePath string, optionMounts []specs.Mount, bindFiles map[string]string, builtinVolumes, volumeMounts []string, shmSize string, namespaceOptions define.NamespaceOptions, secrets map[string]string, sshSources map[string]*sshagent.Source, runFileMounts []string) (runMountTargets []string, agents []*sshagent.AgentServer, sshAuthSock string, err error) {
```

Comment:
Since there looks like the amount of options might expand and the return object might increase, should we create a return struct?

Reply:
YES!  I always get leary when a function takes or returns more than 3 or 4 params, too easy to mess up when calling.

Reply:
would it be a better idea to return a function that cleanups all these mounts? since the only reason we're returning them is to defer the cleanup later? or do you think a struct is better for this case

Reply:
If the defer works for you, that's fine.  My deal is just in general I don't like to see 3 or 4 params being passed back or forth.  YMMV.